src/init_parsing.jl

```diff
@@ -13,6 +13,54 @@ function apply_types_from_mapping!(schema, header, settings, header_provided)
     end
 end
 
+function _startswith(s::AbstractVector{UInt8}, soff::Integer, prefix::AbstractVector{UInt8})
+    length(s) - soff < length(prefix) && return false
+    @inbounds for i in eachindex(prefix)
+        s[i + soff] == prefix[i] || return false
+    end
+    return true
+end
+_startswith(s::AbstractVector{UInt8}, prefix::AbstractVector{UInt8}) = _startswith(s, 0, prefix)
+_startswith(s, soff, prefix::Nothing) = false
+_startswith(s, prefix::Nothing) = false
+
+function skip_rows_init!(lexer_state, parsing_ctx, options, rows_to_skip, comment)
+    i = 0
+    input_is_empty = lexer_state.last_newline_at == UInt(0)
+    lines_skipped_total = 0
+    @inbounds while !input_is_empty
```

Comment:
since we're not actually relying on `input_is_empty` changing in order to break out of the loop, perhaps clearer as
```suggestion
    input_is_empty && return lines_skipped_total
    @inbounds while true
```